### DISCUSSION TITLE
Maximize Audio Volume When Starting A Session

### DISCUSSION THREAD

**Discussion by Chase-Ehlert** [LizardByte/community#228] [Sunshine]
I have a Moonlight and Sunshine setup and my PC stays at the current volume when I launch a session. Because I use a headset at my PC for audio, the volume is pretty low. But because it's so low it is too difficult to hear anything on the client side during a session. The only workaround that I've found is simply maxing out the volume beforehand but I think that could become a headache when I go to use my PC and forget to lower the volume back down to a safe level. I have a lot of programming experience and I was wondering if there was a simple script I could write that would max out the volume when I startup a streaming session and then bump it back down when the session ends. Any ideas?

**Comment by LachlanStuart** [comment]
I have the opposite problem - my Windows Sunshine box always resets to full volume every time I connect, and for some reason that equates to 2x to 4x the volume of anything else on my MacOS Moonlight box. As MacOS doesn't have per-app volume control the only way I can stop it from being deafeningly loud is by manually lowering the "Name Not Available" items in Windows' volume mixer every time I connect.

**Reply by LachlanStuart** [reply]
I just found a solution that might work for your case as well. [SoundVolumeView](https://www.nirsoft.net/utils/sound_volume_view.html) is a command-line tool for setting volumes. In Sunshine's "Advanced" config there's a "Command Preparations" section at the bottom for running things before you connect & after you disconnect. 

I've only tested lowering the volume, but you could try this:
Do command: `c:\<path to where I put SoundVolumeView>\SoundVolumeView.exe /SetVolume sunshine.exe 100` - which should increase the "Name Not Available" item in the mixer that represents Sunshine to 100%.

If that doesn't work for you, there's also `/SetVolume AllAppVolume 100` and a few other possibilities listed in SoundVolumeView's readme.

Once you have that sorted, create an Undo command that resets the volume to the level back to a safe level afterwards.

[a comment by an automated account is not shown]
